keanu-project/src/main/proto/io/improbable/keanu/BayesianNetwork.proto

```diff
@@ -0,0 +1,38 @@
+syntax = "proto3";
+
+package keanu;
+
+option java_package = "io.improbable.keanu";
+option java_outer_classname = "KeanuSavedBayesNet";
+
+message BayesNet {
+    repeated Vertex vertices = 1;
+}
+
+message Vertex {
+    uint32 id = 1;
+    string label = 2;
+    string vertexType = 3;
+    VertexValue value = 4;
```

Comment:
do we need an `isObserved` field?

Reply:
I guess this depends on whether we are separating data and model structure definition. If we are then `isObserved` should probably be in with the data.

Reply:
We thought this should live in the "data layer" rather than the model description layer - it does mean until this comes in we cant save observations but I think thats ok.

Reply:
That makes sense, although I think I don't yet understand why `value` must be part of the description layer.

Reply:
I've been on the fence about this for a bit and have had some good chats with a few people to inform my thinking. I'm currently favouring having a totally separate topology definition, then having additional data layers that are essentially k-v pairs that define data of different types. Migwellian and I just had a chat and we came up with a few different types of data that could potentially occupy separate layers. Interested in other people's thoughts: 
- constants ('real' constants and model defaults that users probably shouldn't consider changing)
- priors (latent variable hyperparameter values)
- observations (data used for model calibration)
- state (the specific values of latent variables at any point during an inference algorithm run, potentially stored for fault tolerance, etc.)